### ISSUE TITLE
memory leak introduced in 1.11.2 and still in master

### ISSUE THREAD

**Issue by soumagne** [ofiwg/libfabric#6506]
Looks like the following leak was recently introduced (trace below is with current master (b64322854a454918948600f7ba1112c192a676de). I started seeing with it 1.11.2, it was not present in 1.11.1 and it is still in master.

```shell
Direct leak of 40 byte(s) in 1 object(s) allocated from:
    #0 0x7ff1fc011837 in __interceptor_calloc (/lib64/libasan.so.6+0xb0837)
    #1 0x7ff1fbd41cff in ofi_mem_init src/mem.c:75
    #2 0x7ff1fbd2b5b6 in fi_ini src/fabric.c:667
    #3 0x7ff1fbd2c0fb in fi_getinfo_ src/fabric.c:967
    #4 0x7ff1fbe97c04 in na_ofi_getinfo /home/jsoumagne/workspace/mercury-hpc/mercury/src/na/na_ofi.c:1705
    #5 0x7ff1fbe9b858 in na_ofi_check_protocol /home/jsoumagne/workspace/mercury-hpc/mercury/src/na/na_ofi.c:3479
    #6 0x7ff1fbe6f7df in NA_Initialize_opt /home/jsoumagne/workspace/mercury-hpc/mercury/src/na/na.c:277
    #7 0x410575 in NA_Test_init /home/jsoumagne/workspace/mercury-hpc/mercury/Testing/na/na_test.c:433
    #8 0x407dd7 in HG_Test_init /home/jsoumagne/workspace/mercury-hpc/mercury/Testing/mercury_test.c:410
    #9 0x404961 in main /home/jsoumagne/workspace/mercury-hpc/mercury/Testing/test_server.c:134
    #10 0x7ff1fb85a041 in __libc_start_main (/lib64/libc.so.6+0x27041)
```

**Comment by shefty** [comment]
Did you configure the build in any particular way?

**Comment by soumagne** [comment]
I don't think there's anything particular about it but here is my configure line:
```
./configure --enable-debug --disable-usnic --disable-mrail --disable-rstream --disable-perf \
  --disable-efa --disable-psm2 --disable-psm --disable-verbs --disable-shm --disable-static \
  --disable-silent-rules
```

**Comment by shefty** [comment]
PRs #6508 and #6509 should fix this in master and v1.11.x branch.  The leak is the result of a double initialization and should be minor.

**Comment by soumagne** [comment]
thanks @shefty for the quick fix! Confirmed the leak is gone.
